### PR TITLE
RFC: Idea on how to handle board dts/kconfig issues

### PR DESCRIPTION
**Pull request by galak**
This is an attempt at an idea on how to decouple Kconfig & dts.  Right now the dts files have the following Kconfig symbols that they utilize:

```
CONFIG_BOOTLOADER_MCUBOOT
CONFIG_BT
CONFIG_CODE_HYPERFLASH
CONFIG_CODE_ITCM
CONFIG_CODE_QSPI
CONFIG_FS_FLASH_MAP_STORAGE
CONFIG_FS_FLASH_STORAGE_PARTITION
CONFIG_MCUMGR_SMP_UART
CONFIG_MODEM_WNCM14A2A
CONFIG_UART_1_NRF_UARTE
CONFIG_UART_MCUX_2
CONFIG_USB_UART_CONSOLE
CONFIG_XIP
```

So one idea, would be to introduce some combination of overlays & cfg in boards dirs.  The change in this example removed CONFIG_BT from the frdm_k64f board.

### CODE REVIEW

**Comment by erwango** [comment]
One initial question:
What was the purpose of these flags initially? (Not sure they all targeted the same goal)

Can't some just be removed and let dts section under these be always there (as this should mostly represent HW which is always there). Then SW using of this HW is conditioned by Kconfig, but this is happening in C code.
I don't think there is any issue if dts section is always present. It's basically like when you have HW that you don't use.

File: boards/arm/frdm_k64f/frdm_k64f_bt.dts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#include <frdm_k64f.dts>
+
+/ {
+	chosen {
+		zephyr,bt-uart = &uart3;
```
Comment (erwango) [inline boards/arm/frdm_k64f/frdm_k64f_bt.dts]: What is the HW which is being described here?
Is that a kind of shield?

File: boards/arm/frdm_k64f/frdm_k64f_bt.dts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#include <frdm_k64f.dts>
```
Comment (erwango) [inline boards/arm/frdm_k64f/frdm_k64f_bt.dts]: How do we select which of frdm_k64f or frdm_k64f_bt should be used in an application?

File: boards/arm/frdm_k64f/frdm_k64f_bt.dts (hunk 1 above)
Comment (galak) [reply boards/arm/frdm_k64f/frdm_k64f_bt.dts]: in this case its a shield

File: boards/arm/frdm_k64f/frdm_k64f_bt.dts (hunk 2 above)
Comment (galak) [reply boards/arm/frdm_k64f/frdm_k64f_bt.dts]: same way we would any board.

File: boards/arm/frdm_k64f/frdm_k64f_bt.dts (hunk 1 above)
Comment (MaureenHelm) [reply boards/arm/frdm_k64f/frdm_k64f_bt.dts]: So why not actually do this as a shield in `boards/shields`? There are at least two boards (frdm_k64f and mimxrt1050_evk) that could use it

Comment (galak) [reply boards/arm/frdm_k64f/frdm_k64f_bt.dts]: For frdm_k64f & mimxrt1050_evk, what is the "shield" board for bluetooth?  Is this a FRDM-KW41Z or something else?

**Comment by nashif** [comment]
I think this will not scale. Fact is, frdm_k64f does not have bluetooth support, so there should not be anything about bluetooth in the board definition. This should be addressed using shields for example.

File: boards/arm/frdm_k64f/frdm_k64f_bt.dts (hunk 1 above)
Comment (MaureenHelm) [reply boards/arm/frdm_k64f/frdm_k64f_bt.dts]: Yes, FRDM-KW41Z can be used either as a base board or as a shield.

Comment (galak) [reply boards/arm/frdm_k64f/frdm_k64f_bt.dts]: > Yes, FRDM-KW41Z can be used either as a base board or as a shield.

Right, I guess I was asking, is the assumption that it is specifically a KW41Z as a shield for CONFIG_BT on FRDM-K64F & mimxrt1050_evk.  Or the assumption more generic about some random BT device/shield?

**Pull request closed without merging**